### ISSUE TITLE
Script ram usage miscalculation

### ISSUE THREAD

**Issue by Arokha** [danielyxie/bitburner#4070]
Bitburner v2.0.2 (3067703c)
```js
/** @param {NS} ns */
export function getPlayerEx(ns){
	let base = ns.getPlayer();
	
	// 'of' because array
	base.factionInfo = {};
	for(const faction of base.factions){
		base.factionInfo[faction] = {
			rep: ns.singularity.getFactionRep(faction),
			favor: ns.singularity.getFactionFavor(faction)
		}
	}

	// 'in' because object
	base.companyInfo = {};
	for(const corporation in base.jobs){
		base.companyInfo[corporation] = {
			rep: ns.singularity.getCompanyRep(corporation),
			favor: ns.singularity.getCompanyFavor(corporation)
		}
	}

	return base;
}
```

This function does invoke any methods in the ns.corporation namespace, and yet:

![image](https://user-images.githubusercontent.com/15028025/187833536-67dd6a28-93f1-46de-9c34-2887ba021718.png)

So I would imagine the parser/lexer/thingy for calculating what methods I'm using is assuming because I have a variable named `corporation` it's referring to the NS1 construct or something of the same name, despite being in NS2? Or something?

I can just rename the variable to get around it, just putting a thing here since I didn't notice any issues with 'namespace' or 'corporation' in them.

**Comment by Snarling** [comment]
You're correct, the static ram parser is just checking for the keywords that have ramcosts assigned to them. You would need to rename your variable something other than corporation in order to avoid the cost.

**Comment by hydroflame** [comment]
Kindof impossible to solve.
